engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -117,40 +115,45 @@ struct blessed_hammer_tick_t : public paladin_spell_t
   void impact( action_state_t* s ) override
   {
     paladin_spell_t::impact( s );
-
     // apply BH debuff to target_data structure
-    td( s -> target ) -> debuff.blessed_hammer -> trigger();
+    // Does not interact with vers.
+    // To Do: Investigate refresh behaviour
+    td( s -> target ) -> debuff.blessed_hammer -> trigger(
+      1,
+      s -> attack_power * p() -> talents.blessed_hammer -> effectN( 1 ).percent()
+    );
   }
 };
 
 struct blessed_hammer_t : public paladin_spell_t
 {
   blessed_hammer_tick_t* hammer;
-  int num_strikes;
+  double num_strikes;
 
   blessed_hammer_t( paladin_t* p, const std::string& options_str ) :
     paladin_spell_t( "blessed_hammer", p, p -> talents.blessed_hammer ),
     hammer( new blessed_hammer_tick_t( p ) ), num_strikes( 2 )
   {
-    add_option( opt_int( "strikes", num_strikes ) );
+    add_option( opt_float( "strikes", num_strikes) );
     parse_options( options_str );
 
     // Sane bounds for num_strikes - only makes three revolutions, impossible to hit one target more than 3 times.
     // Likewise calling the spell with 0 or negative strikes is sort of pointless.
-    if ( num_strikes < 1 )
+    if ( num_strikes <= 0 )
     {
       num_strikes = 1;
-      sim -> error( "{} trying to hit less than one time with blessed_hammer, value changed to 1", p -> name() );
-    }
-    if ( num_strikes > 3 )
-    {
-      num_strikes = 3;
-      sim -> error( "{} trying to hit more than three times with blessed_hammer, value changed to 3", p -> name() );
+      sim -> error( "{} invalid blessed_hammer strikes, value changed to 1", p -> name() );
     }
+    // if ( num_strikes > 3 )
```

Comment:
You should probably keep some sort of upper bound for the number of ticks, even if you want to make it more flexible for the user.